File: commons/test-utils/src/test/java/com/cedarsoft/test/utils/LocaleExtensionTest.java
package com.cedarsoft.test.utils;

import static org.assertj.core.api.Assertions.*;

import java.util.Locale;

import org.junit.jupiter.api.*;

/**
 * @author Johannes Schneider (<a href="mailto:js@cedarsoft.com">js@cedarsoft.com</a>)
 */
@WithLocale("ja_JP")
public class LocaleExtensionTest {
  @Test
  void testWithout() {
    //Value from "global" config for complete test
    assertThat(Locale.getDefault()).isEqualTo(Locale.JAPAN);
  }

  @WithLocale("de_DE")
  @Test
  void testDe() {
    assertThat(Locale.getDefault()).isEqualTo(Locale.GERMANY);
  }

  @WithLocale("en_US")
  @Test
  void testUs() {
    assertThat(Locale.getDefault()).isEqualTo(Locale.US);
  }
}

File: commons/test-utils/src/test/kotlin/it/neckar/open/test/utils/LocaleExtensionTest.kt
package it.neckar.open.test.utils

import org.assertj.core.api.Assertions.assertThat
import org.junit.jupiter.api.Test
import java.util.Locale

/**
 */
@WithLocale("ja_JP")
class LocaleExtensionTest {
  @Test
  fun testWithout() {
    //Value from "global" config for complete test
    assertThat(Locale.getDefault()).isEqualTo(Locale.JAPAN)
  }

  @WithLocale("de_DE")
  @Test
  fun testDe() {
    assertThat(Locale.getDefault()).isEqualTo(Locale.GERMANY)
  }

  @WithLocale("en_US")
  @Test
  fun testUs() {
    assertThat(Locale.getDefault()).isEqualTo(Locale.US)
  }
}
